contribute/guidelines_for_iron_developers.rst

```diff
@@ -0,0 +1,100 @@
+========================================
+Guidelines for OpenCMISS-Iron Developers
```

Comment:
This page is not explicitly linked to the website. As you suggest, this page can be removed once the any relevant information has been removed. Created a new github issue for this: https://github.com/OpenCMISS/documentation/issues/77